tests/test_git_init.py

```diff
@@ -2,49 +2,31 @@
 
 import pathlib
 import subprocess
+import typing
 
 import pytest
 
 
 @pytest.mark.parametrize(
-    "git_init_cookiecutter_option",
-    ["initialise_git_repository=True", "initialise_git_repository=False"],
+    "git_init",
+    [{"initialise_git_repository": True}, {"initialise_git_repository": False}],
 )
 def test_initialisation_of_git_repo(
+    generate_package: typing.Callable,
     tmp_path: pathlib.Path,
-    project_config: dict,
-    git_init_cookiecutter_option: str,
+    git_init: dict[str, bool],
 ) -> None:
-    """
-    Checks to see if git was correctly initialised if desired.
-
-    Args:
-        tmp_path: A temporary directory path object which is unique.
-        project_config: A dictionary with values for the cookiecutter template,
-            as defined in the cookiecutter.json
-        git_init_cookiecutter_option: A string defined in cookiecutter.json,
-            which determines if git should be initialised or not
-
-    """
+    """Checks to see if git was correctly initialised if desired."""
+    test_config = {
+        "github_username": "test-user",
+        "project_short_description": "description",
+        "project_name": "Cookiecutter Test",
+        **git_init,
```

Comment:
And what you're improving above in explicitly naming the variable in the decorator, you're losing here by doing a 'blind' dictionary expansion which doesn't show the key/value pairs being passed into the `test_config` dict.